examples/coap-server.c

```diff
@@ -336,9 +342,17 @@ init_resources(coap_context_t *ctx) {
 
 static void
 fill_keystore(coap_context_t *ctx) {
-  static uint8_t key[] = "secretPSK";
-  size_t key_len = sizeof( key ) - 1;
-  coap_context_set_psk( ctx, "CoAP", key, key_len );
+  if (cert_file) {
+    coap_dtls_pki_t dtls_pki;
+    memset (&dtls_pki, 0, sizeof(dtls_pki));
+    dtls_pki.public_cert = cert_file;
+    dtls_pki.private_key = cert_file;
+    coap_context_set_pki(ctx, &dtls_pki);
+    if (key_defined)
+      coap_context_set_psk(ctx, "CoAP", key, key_length);
```

Comment:
As the key can change, the psk_identity ("user") should be modifiable as well.

I also wonder if the code can be de-duped by putting the `if (key_defined)...` after the `if (certfile)` — when no certificate is provided, the PSK mode would be used only if a key was provided (cf. deprecation of the hard-wired default key i have commented elsewhere).

Reply:
I will add in configurable identities.  Code otherwise reads as

````
static void
fill_keystore(coap_context_t *ctx) {
  if (cert_file) {
    coap_dtls_pki_t dtls_pki;
    memset (&dtls_pki, 0, sizeof(dtls_pki));
    dtls_pki.public_cert = cert_file;
    dtls_pki.private_key = cert_file;
    coap_context_set_pki(ctx, &dtls_pki);
    if (key_defined)
      coap_context_set_psk(ctx, "CoAP", key, key_length);
  }
  else if (key_defined) {
    coap_context_set_psk(ctx, "CoAP", key, key_length);
  }
  else if (coap_dtls_is_supported() || coap_tls_is_supported()) {
    coap_log(LOG_DEBUG,
             "(D)TLS not enabled as neither -k or -c options specified\n");
  }
}
````